Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+        # all possible pandas kwargs will be passed to __init__, where
+        # pandas will be able to process them
+        if not args and (not kwargs or all_kwargs_are_pandas):
+            return super().__new__(cls)
+
+        # pandas can call this constructor (through Table._constructor) internally
+        # when doing transformations: its signatures are either a BlockManager or dict as the sole arg,
+        # or a ndarray as the sole arg, accompanied by at least one kwarg
+        # this serves the purpose of transforming the first arg into its matching kwarg,
+        # which will then be processed by the if clause a couple of lines above this
+        if len(args) == 1 and (isinstance(args[0], (pd.core.internals.BlockManager, dict))
+                               or (isinstance(args[0], np.ndarray) and len(kwargs) != 0 and all_kwargs_are_pandas)):
+            return cls(data=args[0], **kwargs)
+        ##### END PANDAS SUBCLASS COMPATIBILITY SECTION #####
+
+        if 'filename' in kwargs:
+            args = [kwargs.pop('filename')]
+
+        if isinstance(args[0], str):
+            if args[0].startswith('https://') or args[0].startswith('http://'):
+                return cls.from_url(args[0], **kwargs)
+            else:
+                return cls.from_file(args[0], **kwargs)
+        elif isinstance(args[0], TableBase):
+            return cls.from_table(args[0].domain, args[0])
+        elif isinstance(args[0], pd.DataFrame):
+            return cls.from_dataframe(None, args[0], **kwargs)
+        elif isinstance(args[0], Domain):
+            domain, args = args[0], args[1:]
+            if not args:
+                return cls.from_domain(domain, **kwargs)
+            if isinstance(args[0], TableBase):
+                return cls.from_table(domain, *args)
+            elif isinstance(args[0], list):
+                return cls.from_list(domain, *args)
+            elif isinstance(args[0], pd.DataFrame):
+                return cls.from_dataframe(domain, args[0], **kwargs)
+        else:
+            domain = None
+        return cls.from_numpy(domain, *args, **kwargs)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the object.
+
+        Sets the name and other attributes and calls pandas' initialization.
+        Filters the domain: retains only the columns actually in the table.
+        Sets the weights if they are not set already.
+        """
+        # see the comment in __new__ for the rationale here
+        # also, another tidbit is that pandas has some internals that need to be set up
+        # and expects its arguments to be set appropriately
+        # because we override the constructor arguments in a completely incompatible way,
+        # we need to pass ourselves as the data object if we have already set things up
+        # previously by e.g. creating an empty Table via the __new__ hack in from_X
+        # functions, then filling up with columns.
```

Comment:
Imagine calling `Table(ndarray)`. This goes to `from_numpy`, which creates an empty `Table` with `Table()`. No problem there. Then `from_numpy` fills up that table with columns and a domain and returns it, the same is returned from our `__new__`. Our `__init__` is then called with the sole arg of `ndarray`, but we don't want that. Our table is already constructed, pandas would then break some things internally when it would try to initialize by itself. Therefore, we specify ourselves as the `data` kwarg to tell `pandas` we're initializing from existing data (because it's a subclass of `DataFrame`) and it won't do any damn magic. This ties in with the kwarg-only call a couple of lines below - the only thing passed as an arg in pandas internals is `data`, which we've just set manually. 

I hope that explains things. I don't see a huge problem with my code comment, but that's very likely because I spent a couple of days on this. Please kindly advise on any first-class clarifications you might have kindly gained from the explanation above, thank you very much kindly.